src/promptflow-core/promptflow/core/_prompty_utils.py

```diff
@@ -15,6 +16,41 @@
 from promptflow.core._utils import render_jinja_template_content
 
 
+@dataclass
+class PromptyModelConfiguration:
+    api: str
+    configuration: dict
+    parameters: dict
+    response: str = "first"
+    connection: InitVar[Union[str, _Connection]] = None
+
+    def __post_init__(self, connection=None):
+        if connection:
+            self._connection_config = connection
+        else:
+            self._connection_config = copy.copy(self.configuration)
+            if self._connection_config.get("type", None) == "azure_openai":
+                self._connection_config["api_base"] = self._connection_config.get("azure_endpoint", None)
+
+    def get_connection(self):
+        if not hasattr(self, "_connection"):
+            self._connection = get_connection(self._connection_config)
```

Comment:
the resolve logic should postpone to executor

Reply:
fix it